csharp/src/Drivers/Apache/Hive2/HiveServer2Statement.cs

```diff
@@ -123,9 +138,23 @@ protected async ValueTask<Schema> GetSchemaAsync()
             return SchemaParser.GetArrowSchema(response.Schema);
         }
 
-        protected internal int PollTimeMilliseconds { get; } = PollTimeMillisecondsDefault;
+        protected internal int PollTimeMilliseconds { get; private set; } = PollTimeMillisecondsDefault;
+
+        protected internal int BatchSize { get; private set; } = BatchSizeDefault;
+
+        public static class StatementOptions
```

Comment:
Making this `public` seems like the best choice. But if you think it should be private, let me know.

Reply:
Or maybe made available at the `SparkStatement` level?

Reply:
What if we establish a convention of having a static class in appropriate places, like

```
class HiveServer2Statement : AdbcStatement {
    public static class Options {
        // options common to all HiveServer2-derived drivers go here
        public static string PollTimeMilliseconds = "...";
    }
}

class SparkStatement : HiveServer2Statement {
    new public static class Options : HiveServer2Statement.Options {
        // options specific to Spark go here
        // ...
    }
}

```